packages/nouns-contracts/contracts/governance/NounsDAOLogicV2.sol

```diff
@@ -0,0 +1,966 @@
+        } else if (support == 2) {
+            proposal.abstainVotes = proposal.abstainVotes + votes;
+        }
+
+        receipt.hasVoted = true;
+        receipt.support = support;
+        receipt.votes = votes;
+
+        return votes;
+    }
+
+    /**
+     * @notice Admin function for setting the voting delay
+     * @param newVotingDelay new voting delay, in blocks
+     */
+    function _setVotingDelay(uint256 newVotingDelay) external {
+        require(msg.sender == admin, 'NounsDAO::_setVotingDelay: admin only');
+        require(
+            newVotingDelay >= MIN_VOTING_DELAY && newVotingDelay <= MAX_VOTING_DELAY,
+            'NounsDAO::_setVotingDelay: invalid voting delay'
+        );
+        uint256 oldVotingDelay = votingDelay;
+        votingDelay = newVotingDelay;
+
+        emit VotingDelaySet(oldVotingDelay, votingDelay);
+    }
+
+    /**
+     * @notice Admin function for setting the voting period
+     * @param newVotingPeriod new voting period, in blocks
+     */
+    function _setVotingPeriod(uint256 newVotingPeriod) external {
+        require(msg.sender == admin, 'NounsDAO::_setVotingPeriod: admin only');
+        require(
+            newVotingPeriod >= MIN_VOTING_PERIOD && newVotingPeriod <= MAX_VOTING_PERIOD,
+            'NounsDAO::_setVotingPeriod: invalid voting period'
+        );
+        uint256 oldVotingPeriod = votingPeriod;
+        votingPeriod = newVotingPeriod;
+
+        emit VotingPeriodSet(oldVotingPeriod, votingPeriod);
+    }
+
+    /**
+     * @notice Admin function for setting the proposal threshold basis points
+     * @dev newProposalThresholdBPS must be greater than the hardcoded min
+     * @param newProposalThresholdBPS new proposal threshold
+     */
+    function _setProposalThresholdBPS(uint256 newProposalThresholdBPS) external {
+        require(msg.sender == admin, 'NounsDAO::_setProposalThresholdBPS: admin only');
+        require(
+            newProposalThresholdBPS >= MIN_PROPOSAL_THRESHOLD_BPS &&
+                newProposalThresholdBPS <= MAX_PROPOSAL_THRESHOLD_BPS,
+            'NounsDAO::_setProposalThreshold: invalid proposal threshold bps'
+        );
+        uint256 oldProposalThresholdBPS = proposalThresholdBPS;
+        proposalThresholdBPS = newProposalThresholdBPS;
+
+        emit ProposalThresholdBPSSet(oldProposalThresholdBPS, proposalThresholdBPS);
+    }
+
+    /**
+     * @notice Admin function for setting the minimum quorum votes bps
+     * @param newMinQuorumVotesBPS minimum quorum votes bps
+     *     Must be between `MIN_QUORUM_VOTES_BPS_LOWER_BOUND` and `MIN_QUORUM_VOTES_BPS_UPPER_BOUND`
+     *     Must be lower than or equal to maxQuorumVotesBPS
+     */
+    function _setMinQuorumVotesBPS(uint16 newMinQuorumVotesBPS) external {
+        require(msg.sender == admin, 'NounsDAO::_setMinQuorumVotesBPS: admin only');
+        DynamicQuorumParams memory params = getDynamicQuorumParamsAt(block.number);
```

Comment:
Looking at the logic here, all of these would revert when upgrading from V1 -> V2 because `len > 0` would fail.

Reply:
Can you confirm this one? Should be caught by tests.

Reply:
fixed in https://github.com/nounsDAO/nouns-monorepo/commit/11dc46ccf886767298934511db674350131ebf99